front/src/common/RollingStockSelector/SearchRollingStock.tsx

```diff
@@ -0,0 +1,189 @@
+import CheckboxRadioSNCF from 'common/BootstrapSNCF/CheckboxRadioSNCF';
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import { LightRollingStock } from 'common/api/osrdEditoastApi';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { BsLightningFill } from 'react-icons/bs';
+import { MdLocalGasStation } from 'react-icons/md';
+
+interface Filters {
+  text: string;
+  elec: boolean;
+  thermal: boolean;
+}
+
+export function rollingStockPassesEnergeticModeFilters(
+  modes: LightRollingStock['effort_curves']['modes'],
+  { elec, thermal }: Filters
+) {
+  if (elec || thermal) {
+    const effortCurveModes = Object.values(modes).map(({ is_electric: isElec }) => isElec);
+    const hasAnElectricMode = effortCurveModes.includes(true);
+    const hasAThermalMode = effortCurveModes.includes(false);
+    if ((elec && !hasAnElectricMode) || (thermal && !hasAThermalMode)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+type SearchRollingStockProps = {
+  rollingStocks: LightRollingStock[];
+  rollingStockID?: number;
+  setOpenedRollingStockCardId: (rollingStockID: number | undefined) => void;
+  setFilteredRollingStockList: (rollingStocks: LightRollingStock[]) => void;
+  filteredRollingStockList: LightRollingStock[];
+  setIsLoading?: (isLoading: boolean) => void;
+};
+
+const SearchRollingStock = ({
+  rollingStocks,
+  rollingStockID,
+  setOpenedRollingStockCardId,
+  setFilteredRollingStockList,
+  filteredRollingStockList,
+  setIsLoading,
+}: SearchRollingStockProps) => {
+  const { t } = useTranslation('rollingStockEditor');
+
+  const [filters, setFilters] = useState({
+    text: '',
+    elec: false,
+    thermal: false,
+  });
+
+  const searchMateriel = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFilters({ ...filters, text: e.target.value.toLowerCase() });
+    if (setIsLoading) setIsLoading(true);
+  };
+  const toggleFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFilters({ ...filters, [e.target.name]: !filters[e.target.name as 'elec' | 'thermal'] });
+    if (setIsLoading) setIsLoading(true);
+  };
+
+  function rollingStockPassesSearchedStringFilter(
```

Comment:
keep `rollingStockPassesSearchedStringFilter`, `filterRollingStocks`, `rollingStockPassesEnergeticModeFilters` outside of the component, and bunched together as they are related

Reply:
done